### PR TITLE
[SPARK-49406][SQL][TESTS] Add three test cases for `from_avro/to_avro` from the perspective of `GenericRecord`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to add 3 test cases for `from_avro/to_avro` from the perspective of `GenericRecord`.

### Why are the changes needed?
Just to add test cases and help better understand these two functions: `from_avro` & `to_avro`.

### Does this PR introduce _any_ user-facing change?
No, only just supplementary test cases.

### How was this patch tested?
Update existed UT.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @MaxGekk

**Review by MaxGekk — COMMENTED** [review]
cc @gengliangwang

File: connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroFunctionsSuite.scala [hunk 1]
```diff
@@ -371,4 +372,188 @@ class AvroFunctionsSuite extends QueryTest with SharedSparkSession {
...
+      record = datumReader.read(null, decoder)
+    } finally {
+      if (inputStream != null) {
+        inputStream.close()
+      }
+    }
+    record
+  }
+
+  test("GenericRecord serialize/deserialize") {
```
Comment (MaxGekk) [inline connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroFunctionsSuite.scala:412]: Does this test check Spark's code? If not, let's remove it.

File: connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroFunctionsSuite.scala [hunk 2]
```diff
@@ -371,4 +372,188 @@ class AvroFunctionsSuite extends QueryTest with SharedSparkSession {
...
+      assert(readbackPerson1.get(2).toString === person1.get(2))
+
+      val readbackPerson2 = readbacks(1)
+      assert(readbackPerson2.get(0).toString === person2.get(0))
+      assert(readbackPerson2.get(1).asInstanceOf[Int] === person2.get(1).asInstanceOf[Int])
+      assert(readbackPerson2.get(2).toString === person2.get(2))
+    }
+  }
+
+  test("use `to_avro` to read GenericRecord(stored in `struct` datatype)") {
```
Comment (MaxGekk) [inline connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroFunctionsSuite.scala:498]: `to read`? it should write, shouldn't it?

Comment (panbingkun) [reply connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroFunctionsSuite.scala]: I have refactored this code and it seems clearer now. Please help review it in your free time, thanks!

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Comment by panbingkun** [comment]
> +1, LGTM. Merging to master. Thank you, @panbingkun.

Thanks @MaxGekk.

**Pull request closed without merging**